modules/economy-system/commands/shop.js

```diff
@@ -0,0 +1,133 @@
+            return await interaction.reply({
+                content: interaction.client.strings['not_enough_permissions'],
+                ephemeral: true
+            });
+        }
+        const item = await interaction.options.get('item');
+        const price = await interaction.options.getInteger('price');
+        const role = await interaction.options.getRole('role', true);
+        await createShopItem(item['value'], price, role.id, interaction.client);
+        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['itemCreate'], {'%item%': item['value'], '%price%': price, '%role%': role.name}, { ephemeral: true }));
+        interaction.client.logger.info(`[economy-system] The user ${interaction.user.username}#${interaction.user.discriminator} has created the shop item ${item['value']}`);
+        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] The user ${interaction.user.username}#${interaction.user.discriminator} has created the shop item ${item['value']}`);
+    },
+    'buy': async function (interaction) {
+        const itemName = await interaction.options.get('item');
+        const item = await interaction.client.models['economy-system']['Shop'].findOne({
+            where: {
+                name: itemName['value']
+            }
+        });
+        if (!item) {
+            interaction.reply({
+                content: interaction.client.configurations['economy-system']['strings']['notFound'],
+                ephemeral: true
+            });
+        }
+        const user = await interaction.client.models['economy-system']['Balance'].findOne({
+            where: {
+                id: interaction.user.id
+            }
+        });
+        if (user.balance < item.price) return interaction.reply({
+            content: interaction.client.configurations['economy-system']['strings']['notEnoughMoney'],
+            ephemeral: true
+        });
+        balance(interaction.client, interaction.user.id, 'remove', item.price);
+        await interaction.member.roles.add(item.role);
+        createleaderboard(interaction.client);
+        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['buyMsg'], {'%item%': itemName['value']}, { ephemeral: true }));
+        interaction.client.logger.info(`[economy-system] The user ${interaction.user.username}#${interaction.user.discriminator} has buyed the shop item ${itemName['value']}`);
+        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] The user ${interaction.user.username}#${interaction.user.discriminator} has buyed the shop item ${itemName['value']}`);
+    },
+    'list': async function (interaction) {
+        const msg = await createShopMsg(interaction.client);
+        interaction.reply({
+            content: msg,
+            ephemeral: true
+        });
```

Comment:
commands does not work ):
Could you check if it works for you (tries to send empty message)